### DISCUSSION TITLE
How do multiple filtered subscriber behave?

### DISCUSSION THREAD

**Discussion by febus982** [ag2ai/faststream#1565] [Q&A] [answered]
I have a topic/channel for a resource that contains multiple message types and schemas. (i.e. `books` topic => `book.created` and `book.borrowed` events)

I want to make sure the message ordering is handled correctly (no `borrowed` events handled before `created` ones). Can I create 2 handlers in this way or the subscribers will proceed independently from each other?

```python
@subscriber(
    filter=lambda msg: msg.type == "book.created",
)
async def handle_cerated(msg: BookCreated):
    ...

@subscriber(
    filter=lambda msg: msg.type == "book.borrowed",
)
async def handle_cerated(msg: BookBorrowed):
    ...
```

**Comment by Lancetnik** [comment, marked as answer]
Filter logic just allows you to call different handler functions for one message consumption
It is litterally the single subscriber, that iterates over your handlers and tries to find the first suitable one: https://github.com/airtai/faststream/blob/main/faststream/broker/subscriber/usecase.py#L335
